library/core/src/array/mod.rs

```diff
@@ -36,8 +37,18 @@ pub use iter::IntoIter;
 /// // elements to produce is the length of array down there: only arrays of
 /// // equal lengths can be compared, so the const generic parameter `N` is
 /// // inferred to be 5, thus creating array of 5 elements.
-/// let array = core::array::from_fn(|i| i);
+///
+/// let array: [_; 5] = core::array::from_fn(|i| i);
+/// // indexes are:    0  1  2  3  4
 /// assert_eq!(array, [0, 1, 2, 3, 4]);
+///
+/// let array2: [_; 8] = core::array::from_fn(|i| i * 2);
```

Comment:
Then it would be good to mention the type annotation on this on to contrast with the first one.